src/domain/types.ts

```diff
@@ -23,6 +25,6 @@ export interface Identity {
 }
 
 export interface UserRepository {
-    findUser(userId: string): Promise<User>;
+    findUser(userId: string): Promise<Option<User>>;
```

Comment:
I was under the impression the use of funfix was to be reduced. Is there a reason for pulling it on here?

Reply:
I'm not 100% sure on this so happy to change. We have it on the other repos as well so thought it would be good to be consistent. I think having the monads at this level is better thant than consuming code.

Reply:
after chatting with @diversemix and @norisalsaadie, decided that it was ok